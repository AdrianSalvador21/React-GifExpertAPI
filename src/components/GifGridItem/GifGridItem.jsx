import React from 'react';

const GifGridItem = ({id, title, url, img}) => {
  return (
    <div className="card animate__animated animate__fadeIn">
      <img src={url} alt={title} />
      <p>{title}</p>
    </div>
  );
};

export default GifGridItem;
