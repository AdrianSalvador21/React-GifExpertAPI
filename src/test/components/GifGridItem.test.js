import '@testing-library/react';
import {shallow} from "enzyme";
import React from "react";
import GifGridItem from "../../components/GifGridItem/GifGridItem";

describe('TEST GifGridItem', () => {
  const title = 'Un titulo';
  const url = 'http://localhost:4200';
  const wrapper = shallow(<GifGridItem title={title} url={url}/>);


  test('should show component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should have a p with title', () => {
    const p  = wrapper.find('p');
    expect(p.text().trim()).toBe(title);
  });

  test('should have a image with url & alt', () => {
    const img  = wrapper.find('img');
    // console.log(img.html());
    //console.log(img.props());
    //console.log(img.prop('src'));
    expect(img.prop('src')).toBe(url);
    expect(img.prop('alt')).toBe(title);

  });

  test('should have animate_fadeIn class', () => {
    const div  = wrapper.find('div');
    const className = div.prop('className');
    expect(className.includes('animate__animated')).toBe(true);
  });
});
