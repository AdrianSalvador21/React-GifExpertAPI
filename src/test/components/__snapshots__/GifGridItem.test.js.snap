// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEST GifGridItem should show component correctly 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un titulo"
    src="http://localhost:4200"
  />
  <p>
    Un titulo
  </p>
</div>
`;
